app/src/main/java/xyz/jhughes/laundry/adapters/MachineAdapter.java

```diff
@@ -135,24 +145,134 @@ public void onClick(DialogInterface dialog, int which) {
+
+                String notificationKey = roomName + " " + m.getName();
+
+                if (NotificationCreator.notificationExists(notificationKey)) {
+                    listener.postSnackbar(mContext.getString(R.string.reminder_already_set), Snackbar.LENGTH_LONG);
+                } else {
+                    notificationWithDialog(millisInFuture, notificationKey);
+                }
+            } catch (NumberFormatException | StringIndexOutOfBoundsException e) {
+                if (m.getStatus().compareTo("Out of order") != 0) {
+                    listener.postSnackbar(mContext.getString(R.string.machine_not_running), Snackbar.LENGTH_SHORT);
+                } else {
+                    listener.postSnackbar("This machine is offline but may still be functioning. Visit " + m.getName() + " for details.", Snackbar.LENGTH_LONG);
+                }
+            }
+        }
+    }
+
+    public boolean createNotification(Machine m){
         try {
             int minutesInFuture = Integer.parseInt(m.getTime().substring(0, m.getTime().indexOf(' ')));
-            int millisInFuture = minutesInFuture * 60000; //60 seconds * 1000 milliseconds
+            int milliInFuture = minutesInFuture * 60000; //60 seconds * 1000 milliseconds
 
             String notificationKey = roomName + " " + m.getName();
 
             if (NotificationCreator.notificationExists(notificationKey)) {
                 listener.postSnackbar(mContext.getString(R.string.reminder_already_set), Snackbar.LENGTH_LONG);
             } else {
-                fireNotificationInFuture(millisInFuture, notificationKey);
+                mContext.startService(new Intent(mContext, NotificationCreator.class)
+                        .putExtra("machine", notificationKey)
+                        .putExtra("time", milliInFuture));
+                Toast.makeText(mContext, mContext.getString(R.string.alarm_set), Toast.LENGTH_SHORT).show();
             }
-        } catch (NumberFormatException | StringIndexOutOfBoundsException e) {
-            if (m.getStatus().compareTo("Out of order") != 0) {
-                listener.postSnackbar(mContext.getString(R.string.machine_not_running), Snackbar.LENGTH_SHORT);
-            } else {
-                listener.postSnackbar("This machine is offline but may still be functioning. Visit " + m.getName() + " for details.", Snackbar.LENGTH_LONG);
+            return true;
+        } catch (NumberFormatException | StringIndexOutOfBoundsException e){
+            return false;
+        }
+
+    }
+
+    public void waitForMachine(Machine m){
+        //Constructs the dialog to wait for a machine
+        //checks the server while the dialog is open and the app is running in the background
+        final Machine m2 = m;
+        final Handler handler = new Handler();
+        lockListener.onLock();
+        AlertDialog.Builder machineWaitingDialog = new AlertDialog.Builder(mContext);
```

Comment:
Is it possible to make an alert dialog extended class which will clear up this, attaching listeners in the builder, and make the waitForMachine() method's code much smaller (most of it is dialog creation)?